src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+---
+layout: shared/narrow
+published: false
+title: "Setting up a basic service worker"
+description: "This section describes the most basic service worker possible. It shows you how a client gets, or registers, a service worker. It shows you how a service worker prepares to act as a proxy for its clients. Both subjects have more depth than is shown in this section. It's not intended to be comprehensive; it's intended to give you a basic foundation on which to build service worker knowledge."
+authors:
+  - josephmedley
+published_on: 2015-10-01
+updated_on: 2015-10-01
+order: 4
+key-takeaways:
+  tldr:   
+  - "Call <code>register()</code> to get a service worker." 
+  - "Learn to use promises before attempting service workers."
+  - "Service workers have a limited scope and require HTTPS."
+  - "The installation process has two events: install and activate."
+  - "To start proxying, a service worker needs a call to <code>clients.claim()</code> or a user navigation."
+  - "Pass a promise to <code>waitUntil()</code> to indicate success/failure of asynchronous tasks."
+notes:
+  promises:
+    - "<b>Promises</b>&mdash;Notice the use of <code>.then()</code> at the end of the <code>register()</code> function. This is an example of an ECMAScript 2015 construct called a <a href='http://www.html5rocks.com/en/tutorials/es6/promises/'>promise</a>. Service workers make heavy use of promises. If you've never used promises before, you should familiarize yourself with them before trying to implement a service worker."
+  https-only:
+    - "<b>HTTPS Only</b>&mdash;Service workers can do almost whatever they want to HTTPS requests and responses. Since this would make them targets for man-in-the-middle attacks, they must be served over HTTPS. localhost is exempt from the HTTPS only rule, simplifying development and testing."
```

Comment:
You could link to http://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features here to provide more context.